### ISSUE TITLE
Error when archiving app using Amplify Library in App and App Extension.

### ISSUE THREAD

**Issue by alionthego** [aws-amplify/amplify-ios#1558]
I have an iOS app using the Amplify Framework.  I have added the framework using Swift Package Manager.  After adding the framework to the main app, I went to the target for the Share Extension and manually added the amplify libraries in General - Frameworks and Libraries.

The app builds and runs fine without any errors and I'm able to use the Amplify framework in both the app and app extension.  The Main app stores the user password in keychain and the extension will sign in first time to get a token.

The only issue I have is when I try to archive the app to upload to store I get the following message:

`CFBundleIdentifier Collision. There is more than one bundle with the CFBundleIdentifier value 'com.amazonaws.AWSAuthCore' under the iOS application 'AHChat.app'. With error code STATE_ERROR.VALIDATION_ERROR.90685 for id 2818ba85-f436-4a80-a22a-6ee54d8dbb52`

I have been to numerous forums trying to resolve this but I can't.  Even here previously the reply was that Amplify doesn't support app extensions with is absolutely unacceptable.  90 percent of the iOS apps on my device use extensions.  Any app that wants to share media/url's etc needs to use the Share Extension.  Any app that uses notifications beyond simple notifications needs to use the Notification Extension not to mention Watch extension etc...   To simply say we don't support this and make us wait another few years for you to get around to it is unacceptable when it clearly works now as my app demonstrates.  The only issue I have is archiving with the above error and I would appreciate somebody letting me know how I can resolve this.  Thanks.

**Comment by atierian** [comment]
Hi @alionthego, thanks for opening this issue.
Can you please provide some more information that will help us diagnose the issue you're experiencing?

Specifically:
- Amplify version.
- Amplify categories.
- Xcode version.

**Comment by royjit** [comment]
Closing due to inactivity

**Comment by stefano-tilt** [comment]
any updates on this issue or workaround?

**Comment by alionthego** [comment]
hi @stefano-tilt 
I did manage to get it to work eventually.  I abandoned SPM and used Pods.  My pod file was:

```
target 'MyApp' do
  platform :ios, '15.0'
  use_frameworks!

  pod 'Amplify'
  pod 'AmplifyPlugins/AWSAPIPlugin'
  pod 'AmplifyPlugins/AWSS3StoragePlugin'
  pod 'AmplifyPlugins/AWSCognitoAuthPlugin'

end

target 'ShareExtension' do
  platform :ios, '15.0'
  use_frameworks!

end

target 'IntentsExtension' do
  platform :ios, '15.0'
  use_frameworks!

end
```

And then I had to update the build settings of the extensions to include the frameworks.  I did this by manually copying the settings for search paths from the main target to the extensions.  

It works fine with no build errors or issues archiving.  Bear in mind that you still need to login from your extensions as the login token is not shared between frameworks.  I used keychain to keep the password for the user and logged in again for each extension the first time the user used that extensions.  This may not work for you if your app only allows one login session at a time.
